input-output-hk/daedalus
# [DDW-588] Detect User Locale on Daedalus Start

This PR handles the situation where Daedalus starts up and the locale hasn't been set before, as a new user might see. Daedalus will first check if the system locale is Japanese. If the system locale is Japanese, the Daedalus locale defaults to Japanese language until the user confirms the locale via the selection form. If the system locale isn't Japanese, the Daedalus locale defaults to English language until the user confirms the locale via the selection form.

## Screenshots:

- Confirmation it is working where my system preferences are set to **region = Japan** and **language = Japanese**:
 
<img width="1680" alt="language-selection" src="https://user-images.githubusercontent.com/13347886/54334494-725b7580-45e3-11e9-9e88-028375edd2f6.png">

## How to Test:
1. First delete the client side local storage cache generated by `electron-store` which are the contents of this file: `/Users/marcushurney/Library/Application Support/Daedalus/config.json`

2. On macOS, click the Apple icon in the top left corner of the system menu bar, then click System Preferences. Then click Language and Region and change region to Japan. Then click the `+` symbol under Preferred Languages and select Japanese. Click and drag Japanese above your other languages so it becomes your primary language.

3. Restart your computer.

4. Run Daedalus in dev mode to see the language has defaulted to Japanese.

5. To reset your system locale to your original language preference follow the screenshots below, which reset it to English:

<img width="877" alt="select-region" src="https://user-images.githubusercontent.com/13347886/54335252-991aab80-45e5-11e9-89b8-03e3f6c3ffba.png">

<img width="670" alt="restart-button" src="https://user-images.githubusercontent.com/13347886/54335271-a8015e00-45e5-11e9-89b5-a32fceae7f8e.png">


## Review Checklist:

### Basics


### Code Quality

### Testing

### After Review:
